doc/specs/#148 - Repair Support.md

```diff
@@ -0,0 +1,77 @@
+
+### Repair Feature for different Installer Types with Native Repair Capabilities
+  - Msi, Wix : The repair command will use the built-in repair features of the MSI installer type. It will run the msiexec command with the default [repair options](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/msiexec#repair-options) to repair the application using a ShellExecute call or by invoking [MsiReinstallProduct](https://learn.microsoft.com/en-us/windows/win32/api/msi/nf-msi-msireinstallproductw) API call with `REINSTALLMODE` mode property set.
+  - Msix : The repair command will use the built-in repair features of the MSIX installer type. It will make an MSIX API call to register the application package, which will attempt to repair the package.
+    > We can't do the same thing as setting app repair for MSIX app right now because we can't use the private APIs that it uses to do repair operations with winget.
+  - MSStore : The repair command will make an MSStore API  [StartProductInstallAsync](https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.store.preview.installcontrol.appinstallmanager.startproductinstallasync?view=winrt-22621) call to repair the application with 'Repair' property of AppInstallOption set to true.
+
+### Repair Feature for Installer Types that require Custom Repair Switch
+  -  Burn/Exe : The custom switch for repair in the YAML manifest file will be used to perform the repair. The repair command will run the installer with the custom switch to fix the application. To have enough flexibility, different options are possible depending on the installer source used for repair. 
+   - Installed Source: If the YAML manifest file has the 'ModifyRepair' switch, the repair command will use the modify command in the ARP 'ModifyPath' registry key with the repair switch through a ShellExecute call.
+   - Remote Source: If the YAML manifest file has the 'InstallerRepair' switch, the repair command will get the matching installer from the remote source and use the repair switch on the downloaded installer through a ShellExecute call.
+      > If neither switch is specified, the repair command will display an error message.
+
+> Note: The initial implementation will not support repair for Nullsoft, Inno, Portables installer types. Based on feedback from the community, we may add the repair feature for these installer types in a future release.
+
+## Manifest Changes
+Addition of `ModifyRepair` property to InstallerSwitch
+- The `ModifyRepair` property is used to define the custom repair option that works with ModifyPath ARP entry for those installer types that have ModifyPath ARP entry and allow repair through Modify flow.
+
+Addition of `InstallerRepair` property to InstallerSwitch
+-  With the `InstallerRepair` switch, the installer that matches the remote source is downloaded and then repaired by using ShellExecute to call the repair switch on the downloaded installer.
```

Comment:
Also - some packages don't have a dedicated modify exe, but instead use the uninstaller. Take for example `Microsoft.DotNet.DesktopRuntime.6` - where the `UninstallString` is the same as `ModifyPath` This is a case where it has been done "correctly" and the `ModifyPath` has been set but it is possible, and likely probable, that some publisher out there is only using the `UninstallString` for the modify / repair flow. This would make it easy to add `RepairType: Uninstaller` which would call the `UninstallString` with the appropriate switch, rather than requiring a new `InstallerSwitch`

Reply:
To answer the first question, we designed `ModifyRepair `and `InstallerRepair `to enable some flexibility around the source of the installer that can apply the repair switch.

`ModifyRepair `- This switch is mainly for the Modify flow from the settings apps, which in most cases would use a cached local installer to enter the modify flow and repair.

`InstallerRepair `- This switch allows the repair switch to be applied on a matching installer from a remote source. That is, we select a matching installer from the manifest that defines the `InstallerRepair `switch, then download that installer and use it for the repair operation. This adds more value for scenarios such as:

1. We need a complete installer to perform the repair operation and the cached installer on the local machine is incomplete for some reason.
2. The local package cache is missing or corrupted.
3. We want to have some control over the mapping between the installer and the repair switch if this becomes important at some point.

By the way, I like your idea of using a single switch for `Repair`, and a separate `RepairType `field that controls the behavior. This gives more flexibility and extensibility to the existing design, considering we are discovering multiple ways to trigger the repair behavior, such as `Installer, Modify, Uninstaller` etc.

I will update the design document with these feedback.